tests/test_docx_adapter.py

```
@@ -0,0 +1,205 @@
+            )
+        ]
+
+        with patch.object(self.adapter, 'extract_segments', return_value=mock_segments):
+            metadata = self.adapter.collect_metadata("test.docx")
+
+            assert metadata.file_path == "test.docx"
+            assert metadata.format == "docx"
+            assert metadata.word_count == 3
+            assert metadata.character_count == 14
+            assert metadata.segment_count == 2
+            assert metadata.has_headers_footers == False
```

Ruff (0.13.1)

Avoid equality comparisons to `False`; use `not metadata.has_headers_footers:` for false checks

Replace with `not metadata.has_headers_footers`

(E712)